packages/main/src/Menu.ts

```diff
@@ -277,6 +283,25 @@ class Menu extends UI5Element {
 		this._menuItems.forEach(item => {
 			item._siblingsWithIcon = siblingsWithIcon;
 		});
+
+		if (this._newItemsAdded) {
+			if (this._menuItems.length > 0) {
+				const hasSelectedOrFocusedItem = this._menuItems.some(item => item.selected);
```

Comment:
We could find the item, which will receive the focus directly
`const item = this._menuItems.find(item => item.selected) || this._menuItems[0];`
and focus it.

Reply:
Yes, the current version was test only purposes, not a real PR. Now it's time to polish the code.